OpenTrons/opentrons-api
# feature(app, shared-data):wire-up release labware from gripper command

# Overview

closes [EXEC-759](https://opentrons.atlassian.net/browse/EXEC-759) - fix render deck map when trash is fixed (python api version  < 2.16).
part of [EXEC-723](https://opentrons.atlassian.net/browse/EXEC-723) - POST `unsafe/ungripLabware`

## Test Plan and Hands on Testing

upload the following protocol without labware on the deck:
```
from opentrons import protocol_api, types

metadata = {
    "apiLevel": "2.15",
}

requirements = {
    "robotType": "OT-3"
}

def run(ctx: protocol_api.ProtocolContext) -> None:
    tip_rack_50 = ctx.load_labware("opentrons_flex_96_filtertiprack_50ul", 4)
    ctx.move_labware(labware=tip_rack_50, new_location=5, use_gripper=True)
```

1. enter ER
2. select manually remove labware
3. make sure that after the count down the gripper opens the grip.
4. make sure deck map is rendering.

* tested deck map locally - works as expected.
* tested ungrip of the robot - works as expected. 

## Changelog

POST unsafe/ungripLabware when countdown is over.
fixed but when using deck map with fixed trash with python protocols bellow 2.16.
added shared-data method to get fixed trash definition

## Review requests

changes make sense? did I forget anyhing? 

## Risk assessment

low.


[EXEC-759]: https://opentrons.atlassian.net/browse/EXEC-759?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[EXEC-723]: https://opentrons.atlassian.net/browse/EXEC-723?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
